### ISSUE TITLE
Transitioner NullReference

### ISSUE THREAD

**Issue by Symbai** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#845]
Using 2.3.1 version:
~~Randomly when dragging a new control (grid, listbox, textbox, button, doesn't matter) onto my form a nullreference exception in the designer occurs.
I don't drag the controls on the transitioner, it's just enough for it to be on my application at all.~~

Actually the issue happens randomly but quite often. Happens when dragging controls on designer, happens when exiting from debugging. Happens when controls in designer get changed.

```
NullReferenceException: Object reference not set to an instance of an object.

 at MaterialDesignThemes.Wpf.Transitions.Transitioner.ActivateFrame(Int32 selectedIndex, Int32 unselectedIndex)
   at MaterialDesignThemes.Wpf.Transitions.Transitioner.<.ctor>b__14_0(Object sender, RoutedEventArgs args)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseEvent(RoutedEventArgs e)
   at System.Windows.BroadcastEventHelper.BroadcastEvent(DependencyObject root, RoutedEvent routedEvent)
   at System.Windows.BroadcastEventHelper.BroadcastLoadedEvent(Object root)
   at MS.Internal.LoadedOrUnloadedOperation.DoWork()
   at System.Windows.Media.MediaContext.FireLoadedPendingCallbacks()
   at System.Windows.Media.MediaContext.FireInvokeOnRenderCallbacks()
   at System.Windows.Media.MediaContext.RenderMessageHandlerCore(Object resizedCompositionTarget)
   at System.Windows.Media.MediaContext.RenderMessageHandler(Object resizedCompositionTarget)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by olegsavelos** [comment]
This is still happening for me, was this resolved ?

**Comment by Keboo** [comment]
@olegsavelos are you on the latest 2.5.1 release? If not, can you post a reproduction project/steps?

**Comment by olegsavelos** [comment]
Yes on latest build, the exception happens here 

```
  private void ActivateFrame(int selectedIndex, int unselectedIndex)
        {
            if (!IsLoaded) return;

            TransitionerSlide oldSlide = null, newSlide = null;
            for (var index = 0; index < Items.Count; index++)
            {
                var slide = GetSlide(Items[index]);
                if (index == selectedIndex)
                {
                    newSlide = slide;
                    slide.SetCurrentValue(TransitionerSlide.StateProperty, TransitionerSlideState.Current);
                }
                else if (index == unselectedIndex)
                {
                    oldSlide = slide;
                    slide.SetCurrentValue(TransitionerSlide.StateProperty, TransitionerSlideState.Previous);
                }
                else
                {
                    slide.SetCurrentValue(TransitionerSlide.StateProperty, TransitionerSlideState.None);
                }
                Panel.SetZIndex(slide, 0);
            }

            if (newSlide != null)
            {
                newSlide.Opacity = 1;
            }

            if (oldSlide != null && newSlide != null)
            {
                var wipe = selectedIndex > unselectedIndex ? oldSlide.ForwardWipe : oldSlide.BackwardWipe;
                if (wipe != null)
                {
                    wipe.Wipe(oldSlide, newSlide, GetTransitionOrigin(newSlide), this);
                }
                else
                {
                    DoStack(newSlide, oldSlide);
                }

                oldSlide.Opacity = 0;
            }
            else if (oldSlide != null || newSlide != null)
            {
                DoStack(oldSlide ?? newSlide);
                if (oldSlide != null)
                {
                    oldSlide.Opacity = 0;
                }
            }

            _nextTransitionOrigin = null;
        }
```
when calling 

```
   private TransitionerSlide GetSlide(object item)
        {
            if (IsItemItsOwnContainer(item))
                return (TransitionerSlide)item;

            return (TransitionerSlide)ItemContainerGenerator.ContainerFromItem(item);
        }
```
My first thought was Virtualization but even when i tried an non-virtualized panel the effect was the same.

Here is the xaml 

```
     <md:Transitioner Grid.Row="1"
                                         SelectedItem="{Binding SelectedItem}"
                                         ItemsSource="{Binding ItemsView}">
                            <md:Transitioner.Resources>
                                <Style TargetType="md:TransitionerSlide" BasedOn="{StaticResource {x:Type md:TransitionerSlide}}">
                                    <Setter Property="BackwardWipe">
                                        <Setter.Value>
                                            <md:FadeWipe Duration="0:0:0.5" />
                                        </Setter.Value>
                                    </Setter>
                                    <Setter Property="ForwardWipe">
                                        <Setter.Value>
                                            <md:FadeWipe Duration="0:0:0.5" />
                                        </Setter.Value>
                                    </Setter>
                                </Style>
                            </md:Transitioner.Resources>
                            <md:Transitioner.ItemTemplate>
                                <DataTemplate>
                                    <ContentControl Content="{Binding Module.View}" />
                                </DataTemplate>
                            </md:Transitioner.ItemTemplate>
                        </md:Transitioner>
```

**Comment by Keboo** [comment]
@olegsavelos I am unable to reproduce the issue from that code. Are you able to provide an example project that reproduces the exception?

**Comment by olegsavelos** [comment]
[https://we.tl/t-uyC2clyp3I](url)

So it seems its boils down to Visibility.
The exception happens when Transitioner is not yet visible and Selected value is provided.

**Comment by Keboo** [comment]
Fixed by #1242
